include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3509,6 +3511,177 @@ namespace internal
+
+      const auto &tria          = dof_handler_l2.get_triangulation();
+      const auto  degree        = dof_handler_l2.get_fe().degree;
+      const auto  dofs_per_cell = dof_handler_l2.get_fe().n_dofs_per_cell();
+
+      const Utilities::MPI::Partitioner partitioner_h1(
+        dof_handler_h1.locally_owned_dofs(),
+        DoFTools::extract_locally_active_dofs(dof_handler_h1),
+        dof_handler_h1.get_communicator());
+
+      const Utilities::MPI::Partitioner partitioner_l2(
+        dof_handler_l2.locally_owned_dofs(),
+        DoFTools::extract_locally_relevant_dofs(dof_handler_l2),
+        dof_handler_l2.get_communicator());
+
+      // h1 elements have hirarchic numbering, l2 elements have lexiographic
+      // numbering therefore, we need to convert the dof indices
+      const auto h1_to_l2 =
+        FETools::hierarchic_to_lexicographic_numbering<dim>(degree);
+      const auto l2_to_h1 =
+        FETools::lexicographic_to_hierarchic_numbering<dim>(degree);
+
+      using DoFCellIterator =
+        typename DoFHandler<dim, spacedim>::active_cell_iterator;
+
+      std::vector<std::pair<types::global_dof_index, types::global_dof_index>>
+        local_l2_to_h1_dofs;
+      std::vector<std::pair<types::global_dof_index, types::global_dof_index>>
+        local_h1_to_l2_dofs;
+      local_l2_to_h1_dofs.reserve(dof_handler_l2.n_locally_owned_dofs());
+      local_h1_to_l2_dofs.reserve(dof_handler_l2.n_locally_owned_dofs());
+
+      // fill local_l2_to_h1_dofs and local_h1_to_l2_dofs
+      std::vector<types::global_dof_index> dof_indices_h1(dofs_per_cell);
+      std::vector<types::global_dof_index> dof_indices_l2(dofs_per_cell);
+      for (const auto &cell : tria.active_cell_iterators())
+        {
+          if (cell->is_locally_owned() || cell->is_ghost())
+            {
+              DoFCellIterator cell_h1 = {&tria,
+                                         cell->level(),
+                                         cell->index(),
+                                         &dof_handler_h1};
+              DoFCellIterator cell_l2 = {&tria,
+                                         cell->level(),
+                                         cell->index(),
+                                         &dof_handler_l2};
+              cell_h1->get_dof_indices(dof_indices_h1);
+              cell_l2->get_dof_indices(dof_indices_l2);
+
+              // collect l2 to h1
+              for (unsigned int i = 0; i < dof_indices_l2.size(); ++i)
+                if (partitioner_l2.in_local_range(dof_indices_l2[i]))
+                  {
+                    const auto local_l2 =
+                      partitioner_l2.global_to_local(dof_indices_l2[i]);
+                    const auto local_h1 = partitioner_h1.global_to_local(
+                      dof_indices_h1[h1_to_l2[i]]);
+                    local_l2_to_h1_dofs.emplace_back(local_l2, local_h1);
+                  }
```

Comment:
Not needed?

Reply:
I removed the part in [d1d4a0a](https://github.com/dealii/dealii/pull/15209/commits/d1d4a0a0c8c62302e6452291c6cbbbce9eb789de)